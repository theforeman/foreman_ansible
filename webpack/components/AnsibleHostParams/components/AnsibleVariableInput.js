import React from 'react';
import PropTypes from 'prop-types';

import { overrideFieldName } from './AnsibleVariableHelpers';
import UnhideButton from './UnhideButton';
import OverrideButton from './OverrideButton';
import KeyInfoPopover from './KeyInfoPopover';

const AnsibleVariableInput = ({
  fieldDisabled,
  toggleOverride,
  fieldOverriden,
  lookupKey,
  lookupValue,
  fieldValue,
  updateFieldValue,
  toggleHidden,
  fieldHiddenValue,
  formModelField,
  keyErrors,
}) => (
  <div className="input-group">
    <span className="input-group-addon">
      <KeyInfoPopover
        lookupKey={lookupKey}
        hidden={fieldHiddenValue}
        keyErrors={keyErrors}
      />
    </span>
    <textarea
      className={`form-control no-stretch ${
        fieldHiddenValue ? 'masked-input' : ''
      }`}
      rows="1"
      value={fieldValue}
      name={overrideFieldName(lookupKey, 'value', formModelField)}
      onChange={e => updateFieldValue(e.target.value)}
      disabled={fieldDisabled}
    />
    <span className="input-group-btn">
      <UnhideButton
        toggleHidden={toggleHidden}
        hidden={fieldHiddenValue}
        lookupKey={lookupKey}
      />
      <OverrideButton
        toggleField={toggleOverride}
        fieldOverriden={fieldOverriden}
        keyId={lookupKey.id}
      />
    </span>
  </div>
);

AnsibleVariableInput.propTypes = {
  fieldDisabled: PropTypes.bool.isRequired,
  toggleOverride: PropTypes.func.isRequired,
  fieldOverriden: PropTypes.bool.isRequired,
  lookupKey: PropTypes.object.isRequired,
  lookupValue: PropTypes.object.isRequired,
  fieldValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]).isRequired,
  updateFieldValue: PropTypes.func.isRequired,
  toggleHidden: PropTypes.func.isRequired,
  fieldHiddenValue: PropTypes.bool.isRequired,
  formModelField: PropTypes.string.isRequired,
  keyErrors: PropTypes.object.isRequired,
};

export default AnsibleVariableInput;
