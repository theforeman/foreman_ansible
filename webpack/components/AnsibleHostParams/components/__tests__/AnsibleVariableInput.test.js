import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import AnsibleVariableInput from '../AnsibleVariableInput';

import {
  squareLookupKey,
  squareLookupValue,
} from '../../AnsibleHostParams.fixtures';

const props = {
  toggleOverride: () => {},
  updateFieldValue: () => {},
  toggleHidden: () => {},
  formModelField: 'host',
  lookupKey: squareLookupKey,
  lookupValue: squareLookupValue,
  keyErrors: {},
  fieldValue: 'foo',
  fieldHiddenValue: false,
};

const fixtures = {
  'should render': {
    fieldDisabled: false,
    fieldOverriden: false,
    hiddenValue: false,
    ...props,
  },
  'should render when disabled': {
    fieldDisabled: true,
    fieldOverriden: false,
    hiddenValue: false,
    ...props,
  },
  'should render when overriden': {
    fieldDisabled: false,
    fieldOverriden: true,
    hiddenValue: false,
    ...props,
  },
  'should render when hidden': {
    fieldDisabled: false,
    fieldOverriden: false,
    hiddenValue: true,
    ...props,
  },
};

describe('AnsibleVariableInput', () =>
  testComponentSnapshotsWithFixtures(AnsibleVariableInput, fixtures));
