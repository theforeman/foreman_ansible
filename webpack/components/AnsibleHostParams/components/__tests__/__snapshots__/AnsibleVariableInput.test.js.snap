// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleVariableInput should render 1`] = `
<div
  className="input-group"
>
  <span
    className="input-group-addon"
  >
    <KeyInfoPopover
      hidden={false}
      keyErrors={Object {}}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
    />
  </span>
  <textarea
    className="form-control no-stretch "
    disabled={false}
    name="host[lookup_values_attributes][65][value]"
    onChange={[Function]}
    rows="1"
    value="foo"
  />
  <span
    className="input-group-btn"
  >
    <UnhideButton
      hidden={false}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
      toggleHidden={[Function]}
    />
    <OverrideButton
      fieldOverriden={false}
      keyId={65}
      toggleField={[Function]}
    />
  </span>
</div>
`;

exports[`AnsibleVariableInput should render when disabled 1`] = `
<div
  className="input-group"
>
  <span
    className="input-group-addon"
  >
    <KeyInfoPopover
      hidden={false}
      keyErrors={Object {}}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
    />
  </span>
  <textarea
    className="form-control no-stretch "
    disabled={true}
    name="host[lookup_values_attributes][65][value]"
    onChange={[Function]}
    rows="1"
    value="foo"
  />
  <span
    className="input-group-btn"
  >
    <UnhideButton
      hidden={false}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
      toggleHidden={[Function]}
    />
    <OverrideButton
      fieldOverriden={false}
      keyId={65}
      toggleField={[Function]}
    />
  </span>
</div>
`;

exports[`AnsibleVariableInput should render when hidden 1`] = `
<div
  className="input-group"
>
  <span
    className="input-group-addon"
  >
    <KeyInfoPopover
      hidden={false}
      keyErrors={Object {}}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
    />
  </span>
  <textarea
    className="form-control no-stretch "
    disabled={false}
    name="host[lookup_values_attributes][65][value]"
    onChange={[Function]}
    rows="1"
    value="foo"
  />
  <span
    className="input-group-btn"
  >
    <UnhideButton
      hidden={false}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
      toggleHidden={[Function]}
    />
    <OverrideButton
      fieldOverriden={false}
      keyId={65}
      toggleField={[Function]}
    />
  </span>
</div>
`;

exports[`AnsibleVariableInput should render when overriden 1`] = `
<div
  className="input-group"
>
  <span
    className="input-group-addon"
  >
    <KeyInfoPopover
      hidden={false}
      keyErrors={Object {}}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
    />
  </span>
  <textarea
    className="form-control no-stretch "
    disabled={false}
    name="host[lookup_values_attributes][65][value]"
    onChange={[Function]}
    rows="1"
    value="foo"
  />
  <span
    className="input-group-btn"
  >
    <UnhideButton
      hidden={false}
      lookupKey={
        Object {
          "currentOverride": Object {
            "element": "hostgroup",
            "elementName": "first hostgroup",
            "value": false,
          },
          "defaultValue": true,
          "description": "Square param",
          "hiddenValue": false,
          "id": 65,
          "omit": null,
          "override": true,
          "overrideValues": Array [
            Object {
              "id": 26,
              "match": "hostgroup=first hostgroup",
              "omit": false,
              "value": false,
            },
          ],
          "parameter": "square",
          "parameterType": "boolean",
        }
      }
      toggleHidden={[Function]}
    />
    <OverrideButton
      fieldOverriden={true}
      keyId={65}
      toggleField={[Function]}
    />
  </span>
</div>
`;
